### PR TITLE
Additional constraints for space between inserts

### PR DESCRIPTION
**Pull request by goldenapples**
Adds a few additional methods for specifying the required distance between inserts and each other, and distance from the end, both by paragraphs and by word count.

See https://github.com/fusioneng/fusion-theme/issues/3200 for discussion of the logic requirements.

### CODE REVIEW

**Comment by goldenapples** [comment]
I started this branch off of #33 to avoid creating merge conflicts later. The first three commits here aren't really part of this pull request, please don't merge this before that one is finished and landed.

@noppanit can you look this over for me?

File: inc/utils/class-text.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+	/**
+	 * Abstracted helper function for counting the words in a chunk of text.
+	 *
+	 * Given either a string of text or an array of strings, will split it into words and return the word count
+	 *
+	 * @param string|array Text to count words in
+	 * @return int Word count
+	 */
+	public static function word_count( $text ) {
```
Comment (noppanit) [inline inc/utils/class-text.php]: It would be nice to have unit test coverage for this class as well. So we have a nice coverage.

Comment (goldenapples) [inline inc/utils/class-text.php]: Good call. Done in f60ef5f.

Comment (noppanit) [inline inc/utils/class-text.php]: looks good thanks

**Comment by noppanit** [comment]
@goldenapples I think there's a merge conflict but after that resolved @danielbachhuber you want to check again and merge?

**Comment by goldenapples** [comment]
Ok, fixed merge conflicts. Once the build passes on Travis, I think it's good to be merged. @danielbachhuber want to look at it?

One thing I think probably needs to happen is planning out the structure of the args available to `register_speed_bump()`. Those options have grown kind of organically and could use some standardization. Like right now there are options called `minimum_paragraphs_from_end` and `minimum_words_from_end`. That could grow out of hand easily, and it would be good to replace that with something like 

```
'minimum_distance_from_end' => array(
    'paragraphs' => 3,
    'words'      => 75,
    'characters' => null,
),
```

and do the same every time we have a measurement that can be stated in different units...

**Comment by danielbachhuber** [comment]
> One thing I think probably needs to happen is planning out the structure of the args available to register_speed_bump(). Those options have grown kind of organically and could use some standardization.

:100: File an issue for it?

**Pull request merged**